### PR TITLE
Fixes enum generation for TS

### PR DESCRIPTION
**Pull request by rkodev**
## Overview

Resolved Enum Generation for typescript code

Fixes https://github.com/microsoftgraph/microsoft-graph-devx-api/issues/815
Fixes https://github.com/microsoftgraph/microsoft-graph-devx-api/issues/813

## Testing Instructions

Generated code for https://docs.microsoft.com/en-us/graph/api/accessreviewscheduledefinition-post?view=graph-rest-1.0&tabs=typescript

### CODE REVIEW

**Review by zengin — CHANGES_REQUESTED** [review]
Can you please add a test case for this?

**Comment by rkodev** [comment]
> Can you please add a test case for this?

Will do :-)

File: CodeSnippetsReflection.OpenAPI.Test/TypeScriptGeneratorTest.cs [hunk 1]
```diff
@@ -271,5 +284,18 @@ public async Task GenerateAdditionalData()
             Assert.Contains("new ChatMessage", result);
             Assert.Contains("requestBody.from.user.additionalData", result);
         }
+
+        [Fact]
+        public async Task GeneratesEnumsWhenVariableIsEnum()
+        {
+            const string userJsonObject = "{  \"displayName\": \"Test create\",  \"settings\": {    \"recurrence\": {      \"pattern\": {        \"type\": \"weekly\",        \"interval\": 1      },      \"range\": {        \"type\": \"noEnd\",        \"startDate\": \"2020-09-08T12:02:30.667Z\"      }    }  }}";
```
Comment (zengin) [inline CodeSnippetsReflection.OpenAPI.Test/TypeScriptGeneratorTest.cs:291]: nit on readability:
```suggestion
            const string userJsonObject = @"{
    ""displayName"": ""Test create"",
    ""settings"": {
        ""recurrence"": {
            ""pattern"": {
                ""type"": ""weekly"",
                ""interval"": 1
            },
            ""range"": {
                ""type"": ""noEnd"",
                ""startDate"": ""2020-09-08T12:02:30.667Z""
            }
        }
    }
}";
```

**Review by zengin — DISMISSED** [review]

**Review by zengin — APPROVED** [review]
Thank you @rkodev for the tests!

**Pull request merged**